roadmap/implementors-guide/src/node/utility/candidate-validation.md

```diff
@@ -2,12 +2,24 @@
 
 This subsystem is responsible for handling candidate validation requests. It is a simple request/response server.
 
+A variety of subsystems want to know if a parachain block candidate is valid. None of them care about the detailed mechanics of how a candidate gets validated, just the results. This subsystem handles those details.
+
 ## Protocol
 
-Input:
+Input: [`CandidateValidationMessage`](/type-definitions.html#validation-request-type)
 
-- [`CandidateValidationMessage`](/type-definitions.html#validation-request-type)
+Output: [`Statement`](/type-definitions.html#statement-type) via the provided `Sender<Statement>`.
 
 ## Functionality
 
-Given a candidate, its validation code, and its PoV, determine whether the candidate is valid. There are a few different situations this code will be called in, and this will lead to variance in where the parameters originate. Determining the parameters is beyond the scope of this subsystem.
+Given the hashes of a relay parent and a parachain candidate block, and either its PoV or the information with which to retrieve the PoV from the network, spawn a short-lived async job to determine whether the candidate is valid.
+
+Each job follows this process:
+
+- on `PoVOrigin::Network`, send a `QueryPoV` request to the [Availability Store](/node/utility/availability-store.html).
```

Comment:
Based on other review comments, I still think this shouldn't be a thing. But why would this help?

Reply:
Perhaps I'm confused, but my understanding is that:

- in order to validate a parachain block, you need two ingredients: the block itself, and its PoV
- this subsystem has ultimate responsibility for validating parachain blocks

In other words, it's a caller-friendliness thing: if the caller already has the PoV, it can embed that in `PoVOrigin`. If it just has the `CandidateReceipt`, then this module will handle the fetch. The assumption is that we may want to validate candidates from more than one different subsystem, given only the `CandidateReceipt`; by giving Candidate Validation the responsibility to get the PoV in that case, we avoid duplicating that code.

Reply:
Right. The availability store is not networked, though. It just stores data. So if you don't already have all the pieces you'd need, then it couldn't return the candidate.

The availability fetching subsystem hasn't been described yet

Reply:
That's my misunderstanding about the nature of the availability store, then. ac6fabe contains a brief sketch of what candidate fetch might look like.